app/models/nic/interface.rb

```diff
@@ -105,7 +105,7 @@ def normalize_name
       # no hostname was given or a domain was selected, since this is before validation we need to ignore
       # it and let the validations to produce an error
       return if name.empty?
-      if domain.nil? && name.include?('.') && changed_attributes['domain_id'].blank?
+      if Domain.unscoped.find_by(id: domain_id).nil? && name.include?('.') && changed_attributes['domain_id'].blank?
```

Comment:
I think you misunderstood my comment - checking for domain_id was fine, i was referring to the lookups on lines 110 and 113 that may not find properly when scoped.

Reply:
ahhh, gotcha, will update.

Reply:
I guess the controller should be handling if the user doesn't have visibility of the domain, so .unscoped should be fine.